src/graph/FetchEdgesExecutor.cpp

```
@@ -53,55 +54,53 @@ Status FetchEdgesExecutor::prepareClauses() {
         if (!status.ok()) {
             break;
         }
-
-        // Save the type
-        auto iter = colTypes_.begin();
-        for (auto i = 0u; i < colNames_.size(); i++) {
-            auto type = labelSchema_->getFieldType(colNames_[i]);
-            if (type == CommonConstants::kInvalidValueType()) {
-                iter++;
-                continue;
-            }
-            *iter = type.type;
-            iter++;
-        }
     } while (false);
     return status;
 }
 
 Status FetchEdgesExecutor::prepareEdgeKeys() {
     Status status = Status::OK();
-    if (sentence_->isRef()) {
-        auto *edgeKeyRef = sentence_->ref();
+    do {
+        if (sentence_->isRef()) {
+            auto *edgeKeyRef = sentence_->ref();
 
-        srcid_ = edgeKeyRef->srcid();
-        DCHECK_NOTNULL(srcid_);
+            srcid_ = edgeKeyRef->srcid();
+            DCHECK_NOTNULL(srcid_);
```

I don't think you should check null here, maybe you should return an error.